src/main/java/org/sonar/plugins/stash/coverage/CoverageUtils.java

```diff
@@ -0,0 +1,49 @@
+package org.sonar.plugins.stash.coverage;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.sonar.api.measures.CoreMetrics;
+import org.sonar.plugins.stash.StashPluginConfiguration;
+import org.sonar.wsclient.Sonar;
+import org.sonar.wsclient.base.HttpException;
+import org.sonar.wsclient.services.Resource;
+import org.sonar.wsclient.services.ResourceQuery;
+
+public class CoverageUtils {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CoverageUtils.class);
+
+    private CoverageUtils() {}
+
+    public static double calculateCoverage(int linesToCover, int uncoveredLines) {
+        if (linesToCover == 0) {
+            return 100;
+        }
+
+        return (1 - (((double) uncoveredLines) / linesToCover)) * 100;
+    }
+
+    public static Sonar createSonarClient(StashPluginConfiguration config) {
+        String url = config.getSonarQubeURL();
+        String login = config.getSonarQubeLogin();
+        String password = config.getSonarQubePassword();
+        if (login != null) {
+            return Sonar.create(url, login, password);
+        } else {
+            return Sonar.create(url);
+        }
+    }
+
+    public static Double getLineCoverage(Sonar client, String component) {
+        Resource resource = null;
```

Comment:
Currently the try splits the logic in 3 parts. 

When you put all logic inside a try, it improves the readability. Even more, it makes possible to simple extend the logic and extract the block inside try into separated method.

And now, after refactoring, you see I have a question: "Is it possible that "client.find" returns null?". In original solution you cannot void this null check, as you can receive it after throwing exception or after client.find returns null. But if client.find never ever can return null, this check is useless. So after my proposal the code will be:

```java
try {
  Resource resource = client.find(ResourceQuery.createForMetrics(component, CoreMetrics.LINE_COVERAGE_KEY));
  resourse.getMeasureValue(CoreMetrics.LINE_COVERAGE_KEY);
} catch (HttpException e) {
  LOGGER.error("Could not fetch previous coverage for component {}", component, e);
}
```

Much more simple to read and understand and still just 2 lines of code in the try block.